src/fmeval/eval_algorithms/util.py

```diff
@@ -169,149 +170,136 @@ def category_wise_aggregation(dataset: Dataset, score_column_name: str, agg_meth
-    :param model_input: the model input
-    :param model_output: the model output
-    :param model_log_probability: the model log probability
-    :param target_output: the target output
-    :param category: the category
-    :param sent_more_input: the "sent more" input (used by Prompt stereotyping)
-    :param sent_less_input: the "sent less" input (used by Prompt stereotyping)
-    :param sent_more_input_prob: the "sent more" input probability (used by Prompt stereotyping)
-    :param sent_less_input_prob: the "sent less" input probability (used by Prompt stereotyping)
-    :param sent_more_output: the "sent more" output (used by Prompt stereotyping)
-    :param sent_less_output: the "sent less" output (used by Prompt stereotyping)
-
-    IMPORTANT:
-        The attributes of this class MUST match the values of the
-        column name constants in COLUMN_NAMES in src/constants.py.
-
-        Reason:
-        The `from_row` method validates the column names included
-        in its `row` input, making sure that these column names
-        match the attribute names of this class (this validation
-        only occurs for column names that don't correspond to score
-        names).
-
-        Since the `row` input comes from a Ray Dataset produced by
-        the `evaluate` method of an `EvalAlgorithmInterface`, the column
-        names in the row must come from COLUMN_NAMES in src/constants.py.
-
-        Thus, the attribute names of this class must match the constants
-        in COLUMN_NAMES in order for the validation to make sense.
+    This class represents a single record that gets written by the `save_dataset` method.
+    In other words, it represents a single row from the Ray Dataset that is being saved.
+
+    :param scores: A list of EvalScores, where each EvalScore corresponds
+        to one of the score columns in the Ray Dataset being saved.
+    :param non_score_columns: Maps a column name to its contents in the current row
+        (recall that an EvalOutputRecord corresponds to a single Ray Dataset row).
+
+        Note: the keys in `non_score_columns` must belong to constants.COLUMN_NAMES,
+        because constants.COLUMN_NAMES defines which columns are allowed to appear in
+        the saved output, i.e. the schema for an output record.
     """
 
     scores: List[EvalScore]
-    model_input: Optional[str] = None
-    model_output: Optional[str] = None
-    model_log_probability: Optional[float] = None
-    target_output: Optional[str] = None
-    category: Optional[str] = None
-    sent_more_input: Optional[str] = None
-    sent_less_input: Optional[str] = None
-    sent_more_input_prob: Optional[str] = None
-    sent_less_input_prob: Optional[str] = None
-    sent_more_output: Optional[str] = None
-    sent_less_output: Optional[str] = None
-    prompt: Optional[str] = None
-    sent_more_prompt: Optional[str] = None
-    sent_less_prompt: Optional[str] = None
+    non_score_columns: Dict[str, Union[str, float, int]]
```

Comment:
Subjective:
How about we rename this to `dataset_columns`? 
`non_score_columns` is not very intuitive around what it encompasses.

Reply:
I named it `non_score_columns` because the Ray Dataset contains columns for both scores and non-scores (like model output). However, given that we are handling scores in a different manner than all of the other columns (by using `EvalScore`), I think `dataset_columns` is apt. I will make this change.